a2a/auth.go

```diff
@@ -14,8 +14,72 @@
 
 package a2a
 
-// Defines a security scheme that can be used to secure an agent's endpoints.
-// This is a discriminated union type based on the OpenAPI 3.0 Security Scheme.
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// A declaration of the security schemes available to authorize requests. The key
+// is the scheme name. Follows the OpenAPI 3.0 Security Scheme Object.
```

Comment:
nit: can we follow the go documentation style. i.e. "NamedSecuritySchemes is a declaration of ..."?